xdsl/dialects/csl/csl_stencil.py

```diff
@@ -539,21 +536,21 @@ def parse(cls, parser: Parser):
             props["offset_mapping"] = stencil.IndexAttr.get(*offset_mapping)
         parser.parse_punctuation(":")
         res_type = parser.parse_attribute()
-        if isattr(res_type, stencil.StencilTypeConstr):
+        if stencil.StencilTypeConstr.matches(res_type):
             return cls.build(
                 operands=[temp],
                 result_types=[res_type.get_element_type()],
                 properties=props,
             )
-        elif isattr(res_type, base(TensorType[Attribute])):
+        elif AnyTensorTypeConstr.matches(res_type):
```

Comment:
Things like this could be `isa(res_type, TensorType)`. It might be worth thinking about which we prefer